### PR TITLE
Cortex-M: keep the max_pool2d reference off aten's channels-last int8 path

### PR DESCRIPTION
**Pull request by rascani**
### Summary

aten's channels-last max_pool2d buffers each window index in vec::int_same_size_t<opmath_t> and guards it with
TORCH_CHECK(input_depth * input_height * input_width <= numeric_limits<integer_t>::max()) (aten/src/ATen/native/cpu/MaxPoolKernel.cpp, cpu_max_pool_channels_last). For int8 that caps an image at 127 spatial elements -- H*W, with channels not counted. Reproduce with:

  torch.nn.functional.max_pool2d(
      torch.zeros(1, 1, 12, 12, dtype=torch.int8).to(
          memory_format=torch.channels_last), 2, 2)

The Cortex-M reference implementation called F.max_pool2d on the tensor as given, so a channels-last graph tripped that ceiling and raised before producing a result. That is a limit of the eager kernel and not of arm_max_pool_s8, so it only ever struck the host dialect stage -- but it struck it hard, since the whole model test aborts.

Pooling does not depend on the memory format, so the reference now pools a contiguous copy; the existing return already puts the result back in channels-last, and .contiguous() aliases an already-contiguous tensor, so NCHW graphs pay nothing. quantized_max_pool2d_impl is the only reference in this file that pools in the native int8 dtype: avg_pool2d dequantizes to float first and both convolutions promote to int32, so none of them can reach the check.

### Test plan

pytest backends/cortex_m/test/ops/test_max_pool2d.py -- 16 passed, 2 xfailed, dialect and Corstone-300; the new case adds ~2.4 s, in line with the existing implementation cases. Full backends/cortex_m/test dialect run: 404 passed. Reverting input.contiguous() fails the new case with the TORCH_CHECK above.

Authored with assistance from Claude Code.

### CODE REVIEW

File: backends/cortex_m/ops/operators.py [hunk 1]
```diff
@@ -1495,8 +1495,21 @@ def quantized_max_pool2d_impl(
     if ceil_mode:
         raise RuntimeError("quantized_max_pool2d does not support ceil_mode=True")
 
+    # aten's cpu_max_pool_channels_last buffers each window index in
```
Comment (Erik-Lundell) [inline backends/cortex_m/ops/operators.py]: This is a good explanation but it could do with an initial sentence to set the context, it took me some time to understand. I'd suggest starting with:

Max_pool's channels last kernel limits input size, therefore use channels first kernel and convert back to channels last after.

Then go into detail.

File: backends/cortex_m/test/ops/test_max_pool2d.py [hunk 2]
```diff
@@ -34,6 +36,32 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return self.pool(x)
 
 
+class CortexMMaxPool2dPermutedView(torch.nn.Module):
+    """A single-channel NHWC image permuted to NCHW.
+
+    The pool then sees a live view whose strides satisfy plain contiguity
+    while aten still reports it as channels-last, which is the one shape a
```
Comment (Erik-Lundell) [inline backends/cortex_m/test/ops/test_max_pool2d.py]: It is not clear to me what this means for the test. .contiguous is used in the cortex_m impl, does this mean it should fail unless the to cortex_m pass does something?

**Review by Erik-Lundell — COMMENTED** [review]
Minor comments on wording, but looks solid.

I love this wording :)

> it only ever struck the host dialect stage -- but it struck it hard

File: backends/cortex_m/test/ops/test_max_pool2d.py (hunk 2 above)
Comment (rascani) [reply backends/cortex_m/test/ops/test_max_pool2d.py]: Comment updated. This is catching a really weird edge case: if we used `.contiguous()` on the input in `quantized_max_pool2d_impl` this test would fail, whereas `.to(memory_format=contiguous)` succeeds. Only applies when C==1 and channels last input.

**Comment by rascani** [comment]
> Minor comments on wording, but looks solid.
> 
> I love this wording :)
> 
> > it only ever struck the host dialect stage -- but it struck it hard

Claude never misses an opportunity to add some drama.

**Review by AdrianLundell — APPROVED** [review]
